sdk/core/azure-core-amqp/src/amqp/connection_string_credential.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#include "azure/core/amqp/connection_string_credential.hpp"
+
+#include <algorithm>
+#include <azure/core/base64.hpp>
+#include <azure/core/url.hpp>
+#include <iostream>
+#include <iterator>
+#include <unordered_map>
+#include <vector>
+
+#include <azure_c_shared_utility/azure_base64.h>
+#include <azure_c_shared_utility/buffer_.h>
+#include <azure_c_shared_utility/sastoken.h>
+#include <azure_c_shared_utility/strings.h>
+#include <azure_c_shared_utility/urlencode.h>
+
+namespace Azure { namespace Core { namespace _internal { namespace Amqp {
+
+  // Split a source string into separate substrings via a delimiter.
+  std::vector<std::string> SplitString(const std::string& s, char separator)
+  {
+    std::vector<std::string> result;
+
+    const auto len = s.size();
+    size_t start = 0;
+    while (start < len)
+    {
+      auto end = s.find(separator, start);
+      if (end == std::string::npos)
+      {
+        end = len;
+      }
+
+      result.push_back(s.substr(start, end - start));
+
+      start = end + 1;
+    }
+
+    return result;
+  }
+
+  SaslPlainConnectionStringCredential::SaslPlainConnectionStringCredential(
+      const std::string& connectionString)
+      : ConnectionStringCredential(connectionString)
+  {
+    Azure::Core::Url endpointUrl{GetEndpoint()};
+  }
+
+  std::shared_ptr<Network::Transport> SaslPlainConnectionStringCredential::GetTransport() const
```

Comment:
Based on what `GetTransport()` methods do, they rather should be named `CreateTransport()`.

Also, do you need shared_ptr then? If they create transport, unique_ptr is better.

Reply:
A single transport instance can be shared across multiple Connection objects, so shared_ptr is more appropriate.

And at some point, the transport associated with a credential may be cached in the credential, so Get is a more appropriate verb.

Reply:
If I call that method in two different places, I get shared_ptr to two different objects. Should you cache the value that you returned the first time? Then it's truly Get.

Reply:
The design is that eventually the transport object will be cached. This is especially important in the case of SASL based transports.